FKA: The auto-generated label in a <details> element without a <summary> is not keyboard accessible

rniwa opened the pull request:
#### 2115ac47e42743ef7d1178995ffe8968c0bd3878
<pre>
FKA: The auto-generated label in a &lt;details&gt; element without a &lt;summary&gt; is not keyboard accessible
<a href="https://bugs.webkit.org/show_bug.cgi?id=78499">https://bugs.webkit.org/show_bug.cgi?id=78499</a>

Reviewed by Tim Nguyen.

The bug was caused by the summary element in details element&apos;s shadow tree getting skipped during
sequential focus navigation because it has custom focus logic. Fixed the bug by removing this.

The test was originally written by Arko Saha.

* LayoutTests/fast/html/details-no-summary-keyboard-show-hide-expected.txt: Added.
* LayoutTests/fast/html/details-no-summary-keyboard-show-hide.html: Added.
* LayoutTests/fast/shadow-dom/focus-across-details-element-expected.txt: Rebaselined the test. New
behavior matches that of Chrome.
* LayoutTests/fast/shadow-dom/focus-across-details-element.html: Ditto.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/html/HTMLDetailsElement.h:

Canonical link: <a href="https://commits.webkit.org/267899@main">https://commits.webkit.org/267899@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1505d7acb66b6d17b4cb00b12dc6de6ec0174b16

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18000 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18327 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18893 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19833 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16849 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18197 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21623 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18485 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18842 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18216 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18472 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15661 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20708 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15699 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16414 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22949 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16715 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16584 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20817 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17149 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14537 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16249 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/16255 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20609 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2210 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16999 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/fast/html/details-no-summary-keyboard-show-hide.html:
```diff
@@ -0,0 +1,36 @@
+</script>
+<script src="../../resources/js-test-post.js"></script>
+</body>
+</html>
```
This is an orphan closing tag.

rniwa replied:
Oh oops, will add <html>

nt1m commented on LayoutTests/fast/html/details-no-summary-keyboard-show-hide.html:
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="../../resources/js-test-pre.js"></script>
```
Can we stop using the pre/post.js versions?

rniwa replied:
Sure.

rniwa committed:
FKA: The auto-generated label in a <details> element without a <summary> is not keyboard accessible
https://bugs.webkit.org/show_bug.cgi?id=78499

Reviewed by Tim Nguyen.

The bug was caused by the summary element in details element's shadow tree getting skipped during
sequential focus navigation because it has custom focus logic. Fixed the bug by removing this.

The test was originally written by Arko Saha.

* LayoutTests/fast/html/details-no-summary-keyboard-show-hide-expected.txt: Added.
* LayoutTests/fast/html/details-no-summary-keyboard-show-hide.html: Added.
* LayoutTests/fast/shadow-dom/focus-across-details-element-expected.txt: Rebaselined the test. New
behavior matches that of Chrome.
* LayoutTests/fast/shadow-dom/focus-across-details-element.html: Ditto.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/html/HTMLDetailsElement.h:

Canonical link: https://commits.webkit.org/267899@main
